src/CopySignUpSheetToRoles.js

```diff
@@ -87,31 +93,41 @@ class SpeechDetails {
   }
 }
 
+/// Slice a list based on the location of two entries in the list inclusive.
+/// Throws an exception if the resulting slice is empty. Assumes all entires are
+/// unique.
+function between(arr, first, last) {
+  let sliced = arr.slice(arr.indexOf(first), arr.indexOf(last) + 1);
+  if (sliced.length == 0) {
+    throw "sliced list is empty";
+  }
+  return sliced;
+}
+
 class SignUpDetails {
   constructor() {
     const signupsData = getSignUpSheetData();
 
-    this.date = signupsData[DATE_ROW_IDX][DATE_COL_IDX];
+    this.date =
+      signupsData[SIGNUP_ROW_MAP["meetingHeader"]][SIGNUP_DATE_COL_IDX];
     this.meetingLocation =
-      signupsData[MEETING_HEADER_ROW_IDX][MEETING_LOCATION_IDX];
-    this.sergeantAtArms = signupsData[SAA_ROW_IDX][NAME_COL_IDX];
-    this.secretary = signupsData[SECRETARY_ROW_IDX][NAME_COL_IDX];
-    this.toastmaster = signupsData[TMOD_ROW_IDX][NAME_COL_IDX];
-    this.jokemaster = signupsData[JKM_ROW_IDX][NAME_COL_IDX];
-    this.generalEvaluator = signupsData[GE_ROW_IDX][NAME_COL_IDX];
-    this.recorder = signupsData[REC_ROW_IDX][NAME_COL_IDX];
-    this.timer = signupsData[TIM_ROW_IDX][NAME_COL_IDX];
-    this.ahCounter = signupsData[AHC_ROW_IDX][NAME_COL_IDX];
-    this.wordmasterGrammarian = signupsData[GRAM_ROW_IDX][NAME_COL_IDX];
-    this.tableTopicsMaster = signupsData[TTM_ROW_IDX][NAME_COL_IDX];
-    this.speaker1 = new SpeechDetails(signupsData, SPK1_ROW_IDX);
-    this.speaker2 = new SpeechDetails(signupsData, SPK2_ROW_IDX);
-    this.speaker3 = new SpeechDetails(signupsData, SPK3_ROW_IDX);
-    this.evaluator1 = signupsData[EVAL1_ROW_IDX][NAME_COL_IDX];
-    this.evaluator2 = signupsData[EVAL2_ROW_IDX][NAME_COL_IDX];
-    this.evaluator3 = signupsData[EVAL3_ROW_IDX][NAME_COL_IDX];
-    this.waitlistSpeaker1 = new SpeechDetails(signupsData, WLSPK1_ROW_IDX);
-    this.waitlistSpeaker2 = new SpeechDetails(signupsData, WLSPK2_ROW_IDX);
+      signupsData[SIGNUP_ROW_MAP["meetingHeader"]][
+        SIGNUP_MEETING_LOCATION_COL_IDX
+      ];
+
+    // helper function to keep things compact
+    let rows_between = (start, end) => between(SIGNUP_ROW_NAMES, start, end);
+
+    for (let name of rows_between("sergeantAtArms", "tableTopicsMaster").concat(
```

Comment:
nit: adding comments before each of these blocks could be useful (or putting them into helpers)
```
// Add general roles to to signup details
// Add Speaker information to signup details
```